File: ex-12/ex-12_java/Vetor.java
public class Vetor {
  private double dx, dy;

  // Construtora & Destrutora
  public Vetor(double dx, double dy) {
    this.dx = dx;
    this.dy = dy;
  }

  // Getters & Setters
  public double getX() {
    return this.dx;
  }

  public double getY() {
    return this.dy;
  }

  public void setX(double dx) {
    this.dx = dx;
  }

  public void setY(double dy) {
    this.dy = dy;
  }

  // Operadores
  public Vetor add(Vetor v) {
    // Calcula valores do novo vetor
    double dx = this.getX() + v.getX();
    double dy = this.getY() + v.getY();
    // Cria vetor com os valores calculados e retorna
    return new Vetor(dx, dy);
  }

  public Vetor multiply(Vetor v) {
    // Calcula valores do novo vetor
    double dx = this.getX() * v.getX();
    double dy = this.getY() * v.getY();
    // Cria vetor com os valores calculados e retorna
    return new Vetor(dx, dy);
  }
}

File: ex-12/ex-12_python/Vetor.py
class Vetor:
    # Construtora & Destrutora
    def __init__(self, dx, dy):
        self.dx = dx
        self.dy = dy

    # Getters & Setters
    def getX(self):
        return self.dx

    def getY(self):
        return self.dy

    def setX(self, dx):
        self.dx = dx

    def setY(self, dy):
        self.dy = dy

    # Operadores
    def __add__(self, v):
        # Calcula valores do novo vetor
        dx = self.getX() + v.getX()
        dy = self.getY() + v.getY()
        # Cria vetor com os valores calculados e retorna
        return Vetor(dx, dy)

    def __mul__(self, v):
        # Calcula valores do novo vetor
        dx = self.getX() * v.getX()
        dy = self.getY() * v.getY()
        # Cria vetor com os valores calculados e retorna
        return Vetor(dx, dy)
